op-chain-ops/script/cheatcodes_state.go

```diff
@@ -0,0 +1,25 @@
+package script
+
+import (
+	"errors"
+
+	"github.com/ethereum/go-ethereum/core/state"
+
+	"github.com/ethereum-optimism/optimism/op-chain-ops/foundry"
+)
+
+func (c *CheatCodesPrecompile) LoadAllocs(pathToAllocsJson string) error {
+	c.h.log.Info("loading state", "target", pathToAllocsJson)
+	return errors.New("state-loading is not supported")
+}
+
+func (c *CheatCodesPrecompile) DumpState(pathToStateJson string) {
+	c.h.log.Info("dumping state", "target", pathToStateJson)
+	var allocs foundry.ForgeAllocs
```

Comment:
Why do this exactly? We probably want a way to hydrate many tests using these allocs by passing a copy around?

Reply:
After the `Host` is completed running scripts, we still need to export the state to a `core.Genesis` for both verifier and sequencer to use. And to be shared between some tests with the same configuration ideally. Hence I implemented the state DB dump interface, so we can dump directly to a forge allocs / Geth accounts map. The regular geth dump is a different incompatible type.